File: Dataset/Leetcode/train/28/424.cpp
 class Solution {
public:
    int XXX(string haystack, string needle) {
               if (needle == "")
	{
		return 0;
	}
	if (haystack.size() < needle.size())
	{
		return -1;
	}
	for (int i = 0; i < haystack.size()-needle.size()+1; i++)
	{
		if (haystack[i] == needle[0])
		{
			int left = i;
			int right = 0;
			while (haystack[left] == needle[right])
			{
				left++;
				right++;
				if (left == haystack.size() || right == needle.size())
					break;
			}
			if (right == needle.size() )
			{
				return i;
			}

		}
	}
	return -1;
    }
};

File: Dataset/Leetcode/train/111/424.java
class Solution {
    public int XXX(TreeNode root) {
         if(root==null) return 0;
         int leftheight = XXX(root.left);
         int rightheight = XXX(root.right);
         if(leftheight!=0 && rightheight!=0){ 
             return Math.min(leftheight, rightheight)+1;
         }else{ //左子树和右子树可能有一个是空树，这时候需要返回不是空树的那个深度；
                //两个都是空树的话就是一个叶子节点，返回0；
             return leftheight + rightheight +1; 
         }
    }
}

undefined
for (i = 0; i < document.getElementsByTagName("code").length; i++) {     console.log(document.getElementsByTagName("code")[i].innerText); }


